src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListenerImplBase.cs

```diff
@@ -58,7 +58,9 @@ protected HttpDiagnosticListenerImplBase(IApmAgent agent)
 
 		public void OnNext(KeyValuePair<string, object> kv)
 		{
-			Logger.Trace()?.Log("Called with key: `{DiagnosticEventKey}', value: `{DiagnosticEventValue}'", kv.Key, kv.Value);
+			// We only print the key, we don't print the value, because it can contain a http request object, and its ToString prints
+			// the URL, which can contain username and password. See: https://github.com/elastic/apm-agent-dotnet/issues/515
+			Logger.Trace()?.Log("Called with key: `{DiagnosticEventKey}'", kv.Key);
```

Comment:
Maybe it's worth handling #515 in a separate PR? I don't think just not logging information that might be very helpful to us in support cases is the solution. Otherwise we will have to remove a lot of other useful data from logging. For example if I set `SampleAspNetCoreApp` configuration to:
```
{
  "Logging": {
    "LogLevel": {
      "Elastic.Apm": "Trace"
    }
  },
  "ElasticApm": {
    "ServerUrls": "http://my_username:my_password@localhost:8200",
  }
}
```

The logs will contain:
```
{BackendCommUtils} Building HTTP client with BaseAddress: http://my_username:my_password@localhost:8200/ for BackendCommComponentBase (PayloadSenderV2)...
```

```
{PayloadSenderV2} Using the following configuration options: Events intake API absolute URL: http://my_username:my_password@localhost:8200/intake/v2/events, FlushInterval: 10s, MaxBatchEventCount: 10, MaxQueueEventCount: 1000
```

```
{CentralConfigFetcher} Making HTTP request to APM Server... Request: Method: GET, RequestUri: 'http://my_username:my_password@localhost:8200/config/v1/agents?service.name=SampleAspNetCoreApp&service.environment=Production', Version: 2.0, Content: <null>, Headers:
{
}.
```

In a separate PR a possible approach might be to filter logged data (message text and contextual data) for patterns such as credentials in URL and maybe even using patterns configured via `SanitizeFieldNames`.

Reply:
Sure, we can follow up on this. I can leave #515 open and sync during planning on what we would like to do.

I think we addressed the main issue here, but if something is missing, or some more thinking is needed we can definitely follow up on that.